### DISCUSSION TITLE
GitHub Training

### DISCUSSION THREAD

**Discussion by stephen-rowan** [NFT-DAO/Governance-HOLON#79] [Gov-HOLON-GitHub]
## GitHub Training

Welcome to Gov-HOLON GitHub Training

## Blank repositories have been setup for -

https://github.com/Ek-phrasis/Sevs-Kingdom

https://github.com/Ek-phrasis/Troys-Kingdom

## Arrange Zoom Times

https://www.worldtimebuddy.com/

Troy - either 9 to 10 pm NZ time (10 to 11am BST) or 8 to 9 am NZ time (9 to 10pm BST).

Sev - I prefer mornings from 11 to 1pm Berlin Time (10 to 12 am UK ) - but can do early evenings if you need to.

## Training materials 

The training materials are now at ...

https://github.com/NFT-DAO/Governance-HOLON/tree/main/Training

**Comment by stephen-rowan** [comment]
## Hi Troy and Sev,

We could record all or part of our meeting as a dress rehearsal for a short training video. But the first session will be likely be a bit rough and improvised. Lets see what works and change as necessary.

## Lesson

I thought we would start from how we (Gov-HOLON) should be using a GitHub repo if we want to track our work for token rewards later on.

If we can do this, then we treat our business planning and research documents like code. We will be able to track all work we do on them and others will be able to work on them as well.

So we need to start working in Markdown - which is essentially plain text with basic formatting. ( A useful guide to markdown formatting can be found here (https://docs.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax) . As much as possible I will try to refer to GitHub's own guides. We can cover Markdown formatting in another session.

For this lesson we are going to go straight to how to commit work in text format to a repo so we can track changes.

## Practical work - Committing a file

1. OK - go to your repos at the Sandbox organization. An GitHub organization is typically contains all the repos a company or project is working on.

> https://github.com/Ek-phrasis/Sevs-Kingdom
> https://github.com/Ek-phrasis/Troys-Kingdom

2. And find a short word document or pdf you have worked on and select all the text and copy it to the clipboard.

3. Then click on the little pencil on the first page of the repo (called a README.md page). The .md extension stands for Markdown.

4. This will open an edit file page. Paste all the text you have coped from your document into this page. Don't worry how it looks at this stage.

5. Now scroll to the bottom of the page and you will see a Commit Changes section with two dialog boxes

6. Type a title for the text you pasted in the top dialog box

7. Type some comments in the bottom dialog box

8. Then press the green "Commit Changes" button.

9. This will return you to the README.md page - but with the text you have added (don't worry what it looks like).

10. Now look to the top left and you will see a "contributor" link. This tells you how many people are contributing to this file. If you click on it - it will pop a dialog with your name as one of the contributors.

11.  Now to the top right is a "History" link. If you click on that - it will bring up a list of all the changes you have made to the file. These changes are called "commits".

12. Each "commit" will have the title you gave it and you can go back to any point in time to an earlier commit.

13. If you click on the top commit - it will display what changes you have made. And you can also add comments underneath these changes.

14, To return to the main file click on the three dots on the top right and then click on "view file".

15. This takes you to a view of a particular commit or state of the file - indicated by a grey box with the commit number code in it Why it works this way - we will get to later on.

16. For the moment click on the grey box and select "main" from where it says "branches"

17. This will take you to the "main" branch of the file from where you can start editing again.

## Practice

OK - to practice commits

1. Go back to the text you have pasted and try to tidy it up using markdown (we can cover this).
2. Each time you make a lot of changes -  click on the green "Commit Changes" button and add a title and comments if you wish as well.

**Reply by tttroyyy** [reply]
Nice bite sized tutorial. Other tutorials may include: 1) setting up your github profile, 2) how to access previous versions (commits), 3) how to send a push/pull request, 4) how to accept/decline a request, 5) markdown practice, 6) pasting different file types into markdown, 7) merging changes into main branch, 8) setting up a noticeboard (?), 9) adding contributors to a personal repository, 10) contributing to a repository. Just some thoughts based on our conversation. Thanks again.

**Reply by stephen-rowan** [reply]
Great feedback Troy,

I think 1) **setting up your github profile** would be a good next step - as it would allow you and Sev to personalize your own repo space and get ready to work from local forks of repos there.

On the rest ...

2) **how to access previous versions (commits) -** this is relatively straightforward and we could cover in "a detailed look at commits" etc. 
3) **how to send a push/pull request -** I think this would be a byte sized lesson in itself.
4) **how to accept/decline a request -** This could be part of 3) on pull requests.
5) **markdown practice** - Again probably one lesson could cover the most used parts of Markdown
6) **pasting different file types into markdown** - This could form part of 5)
7) **merging changes into main branch** - This would an intermediate lesson - maybe just looking at merge requests.
8) **setting up a noticeboard (?)** - A good place to setup a noticeboard would be GitHub Discussions
9) **adding contributors to a personal repository** - This is quite straightforward and should probably be part of a repo settings lesson.
10) **contributing to a repository** - This is could be a useful overview of repo contribution in general - where to start, how to contribute etc.
